### PR TITLE
Clkmgr testutils refactor

### PR DESCRIPTION
**Pull request by engdoreis**
Fix https://github.com/lowRISC/opentitan/issues/17527

### CODE REVIEW

File: sw/device/lib/testing/clkmgr_testutils.h [hunk 1]
```diff
@@ -20,8 +21,9 @@
 inline bool clkmgr_testutils_get_trans_clock_status(
```
Comment (timothytrippel) [inline sw/device/lib/testing/clkmgr_testutils.h:21]: should this be static? or is it used outside this file too?

File: sw/device/tests/clkmgr_reset_frequency_test.c [hunk 2]
```diff
@@ -74,9 +75,12 @@ bool test_main(void) {
     CHECK_DIF_OK(dif_rstmgr_software_device_reset(&rstmgr));
   } else if (rstmgr_testutils_reset_info_any(&rstmgr, kDifRstmgrResetInfoSw)) {
     LOG_INFO("Back from rstmgr SW reset");
-    CHECK(clkmgr_testutils_check_measurement_enables(&clkmgr,
-                                                     kDifToggleDisabled));
-    CHECK(clkmgr_testutils_check_measurement_counts(&clkmgr));
+    bool all_disabled =
+        CHECK_STATUS_OK(clkmgr_testutils_check_measurement_enables(
```
Comment (timothytrippel) [inline sw/device/tests/clkmgr_reset_frequency_test.c:79]: why is a `CHECK_STATUS_OK` being assigned to a boolean?

File: sw/device/tests/clkmgr_sleep_frequency_test.c [hunk 3]
```diff
@@ -142,8 +142,10 @@ bool test_main(void) {
   CHECK(isr_entered);
 
   // Interrupt happened. Check the measurement state.
-  CHECK(clkmgr_testutils_check_measurement_counts(&clkmgr));
-  CHECK(clkmgr_testutils_check_measurement_enables(&clkmgr, kDifToggleEnabled));
+  CHECK_STATUS_OK(clkmgr_testutils_check_measurement_counts(&clkmgr));
+  bool all_enabled = CHECK_STATUS_OK(
+      clkmgr_testutils_check_measurement_enables(&clkmgr, kDifToggleDisabled));
```
Comment (timothytrippel) [inline sw/device/tests/clkmgr_sleep_frequency_test.c:147]: same here: why is a `CHECK_STATUS_OK` being assigned to a boolean?

File: sw/device/tests/clkmgr_reset_frequency_test.c (hunk 2 above)
Comment (engdoreis) [reply sw/device/tests/clkmgr_reset_frequency_test.c:79]: I also changed the `CHECK_STATUS_OK` macro to unwrap the `status.value` just like the `TRY` macro.
https://github.com/lowRISC/opentitan/blob/04641452e05d96471a8762ddfb19b8a01546da82/sw/device/lib/testing/spi_flash_testutils.c#L144-L149

Please, let me know if you see any issue with that.

File: sw/device/lib/testing/clkmgr_testutils.h (hunk 1 above)
Comment (engdoreis) [reply sw/device/lib/testing/clkmgr_testutils.h]: I made it static.

File: sw/device/tests/clkmgr_reset_frequency_test.c (hunk 2 above)
Comment (timothytrippel) [reply sw/device/tests/clkmgr_reset_frequency_test.c:79]: Sorry where did you change this? I see it for TRY [here](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/base/status.h;drc=a14866dd639e1a8ce24c852adbf83a51f0760c03;l=114), but I don't see anything for `CHECK_STATUS_OK` [here](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/testing/test_framework/check.h;drc=5ac26cd380616758f7919cc467439b6ccc7dcaa5;l=262)?

File: sw/device/lib/testing/test_framework/check.h [hunk 4]
```diff
@@ -258,14 +258,16 @@
  * Prints a human-readable error message if the status represents an error.
  *
  * @param expr An expression which evaluates to a `status_t`.
+ * @return The enclosed OK value.
  */
 #define CHECK_STATUS_OK(expr, ...)                 \
-  do {                                             \
+  ({                                               \
```
Comment (engdoreis) [inline sw/device/lib/testing/test_framework/check.h:264]: Unwrap the returned value.

File: sw/device/tests/clkmgr_reset_frequency_test.c (hunk 2 above)
Comment (engdoreis) [reply sw/device/tests/clkmgr_reset_frequency_test.c:79]: It's in the same commit [here](https://github.com/lowRISC/opentitan/pull/17947/files#r1166451172)

**Comment by pamaury** [comment]
This looks good to me. However, I wonder if having `CHECK_STATUS_OK` return a value is a bit confusing, mostly because of the name. Since it's probably too late to change it (too many places to change), would it make sense to have the version that returns a value be called `UNWRAP_STATUS_OK`? One could then define `CHECK_STATUS_OK` as simply ignoring this value?

File: sw/device/tests/clkmgr_reset_frequency_test.c (hunk 2 above)
Comment (engdoreis) [reply sw/device/tests/clkmgr_reset_frequency_test.c]: I followed Amaury and @cfrantz advise and created a new macro called `UNWRAP`.

File: sw/device/tests/clkmgr_sleep_frequency_test.c (hunk 3 above)
Comment (engdoreis) [reply sw/device/tests/clkmgr_sleep_frequency_test.c:147]: I created a `UNWRAP` macro.

**Review by pamaury — APPROVED** [review]

**Review by timothytrippel — APPROVED** [review]

**Pull request merged**